src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -11,6 +11,8 @@ namespace OpenTelemetry.Metrics;
 
 internal sealed class AggregatorStore
 {
+    // Constant to account for additional space for MetricPointReclaim and a case with no dimensions.
```

Comment:
MetricPointReclaim? It must be overflow attribute, nothing to do with metricpointreclaim

Reply:
Two comments...

* Agree with @cijothomas one of the reservations is only needed when overflow attribute is enabled we should be checking that here.

* The other reservation for no tags/dimensions, why do we treat that as special? Is there a technical reason or perf reason for that? Could we just treat that like any other measurement?

Reply:
its treated specially to make the 0 attribute case the fastest (it can bypass all lookups since we know metricpoints[0] is always for it). The main use case is for those users who already know ahead the keys **and** values, so we ask them to provide it as part of InstrumentationScope(i.e Meter.Tags), and provide nothing as part of measurement reporting time. This somewhat achieves what used to be called "Bounded-Instrument" (something where key/values are known ahead of time) in the old spec. Such a thing maybe coming to the spec (if there is enough interest), but it could take a very long time. Until then the special cased 0 tag case is a reasonable way to achieve that goal.

(Additional Context from other SIG: OTel .NET had this spl. casing from day1. OTel Rust did not have it, and we are actively adding it now to unblock high-perf scenarios)

Reply:
> MetricPointReclaim? It must be overflow attribute, nothing to do with metricpointreclaim

Correct this is for overflow attribute.

Reply:
@cijothomas Could we just maintain the index for the zero-dimension point on `aggregatorStore` to do the cheating? Like...

```csharp
if (tags.Length == 0)
{
    if (!this.zeroDimensionIndex.HasValue)
    {
         // Call normal code which creates or returns a point. 
         // If multiple threads enter here no big deal they'll just all set the same value no need to synchronize I don't think
         this.zeroDimensionIndex = FindMetricPointForTags(tags); 
    }
    return this.zeroDimensionIndex;
}
```

Con: Couple nanos slower than just using 0 once initialized. 
Pro: Doesn't waste memory for metrics not using zero dimensions (don't need to reserve the point).

Reply:
The couple nano secs gain is why I'd prefer to keep it special cased.  The memory wasted is 1 X sizeof(MetricPoint) which should be reasonable.

The main use case for bounded instrument comes from ultra fast use cases, like an app doing network packet routing and such, which are usually measured in nano secs, and we should try to make the metric sdk usable for such needs too.